src/tequila/overlap/overlap.py

```diff
@@ -0,0 +1,88 @@
+from tequila.circuit.circuit import QCircuit, assign_ctrl
+from tequila.circuit.gates import H, X, Y, PauliGate
+from tequila.hamiltonian.qubit_hamiltonian import QubitHamiltonian
+from tequila.objective.objective import ExpectationValue
+
+import numpy as np
+import copy
+
+def make_overlap(U0:QCircuit = None, U1:QCircuit = None)->ExpectationValue:
+    '''
+    Function that calculates the overlap between two quantum states.
+
+    Parameters
+    ----------
+    U0 : QCircuit tequila object, corresponding to the first state.
+         
+    U1 : QCircuit tequila object, corresponding to the second state.
+
+    Returns
+    -------
+    Real and imaginary Tequila objectives to be simulated or compiled.
+
+    '''
+    
+    U_a = copy.deepcopy(U0)
+    U_b = copy.deepcopy(U1)
+    
+    #print(U_a)
+    #print(U_b)
+    
+    ctrl = assign_ctrl(U0=U_a, U1=U_b)
+    
+    #print('Control qubit:',ctrl)
+    
+    U_a.add_controls([ctrl]) #this add the control by modifying the previous circuit
+    U_b.add_controls([ctrl]) #NonType object
+    
+    #bulding the circuit for the overlap evaluation
+    circuit = H(target=ctrl)
+    circuit += U_a
+    circuit += X(target=ctrl)
+    circuit += U_b
+    
+    x = X(ctrl)
```

Comment:
this is tq.gates.X but it should be tq.paulis.X
same for the Y below
easiest way to get around the potential namespace crash when importing is to just do:

x = QubitHamiltonian("X({})".format(ctrl))

or like this:
from tequila import paulis
and then:
x = paulis.X(ctrl)